elasticsearch_collectd.py

```diff
@@ -909,6 +919,26 @@ def load_es_info():
                (ES_VERSION, ES_CLUSTER, ES_MASTER_ELIGIBLE))
 
 
+def detect_es_master():
+    """Determines if this is the current master. This method sets ES_CURRENT_MASTER"""
+    global ES_CURRENT_MASTER
+
+    json = fetch_url("http://" + ES_HOST + ":" + str(ES_PORT) +
+                     "/_nodes/_local")
+    if json is None:
+        ES_CURRENT_MASTER = False
+        log.warning('Unable to determine node \
+information, defaulting to current master %s' % ES_CURRENT_MASTER)
+        return
+
+    # determine current master and update globel setting
+    cluster_state = fetch_url("http://" + ES_HOST + ":" + str(ES_PORT) +
+                            "/_cluster/state/master_node")
+    ES_CURRENT_MASTER = cluster_state['master_node'] == json['nodes'].keys()[0]
+
+    log.notice('current master: %s' % ES_CURRENT_MASTER)
```

Comment:
we should only log if we detect a change in mastership, otherwise you will see every 10s